IOPool/Input/test/testNoParentDictionary.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash -e
+
+function die { echo $1: status $2 ;  exit $2; }
+
+SCRAM_TEST_NAME=TestIOPoolInputNoParentDictionary
+rm -rf $SCRAM_TEST_NAME
+mkdir $SCRAM_TEST_NAME
+cd $SCRAM_TEST_NAME
+
+# Create a new CMSSW dev area
+OLD_CMSSW_BASE=${CMSSW_BASE}
+scram -a $SCRAM_ARCH project $CMSSW_VERSION
+pushd $CMSSW_VERSION/src
+eval `scram run -sh`
+
+# Symlink any libraries and -edmplugincache *except* for the libraries containing the test dictionaries
+for F in ${OLD_CMSSW_BASE}/lib/${SCRAM_ARCH}/* ${OLD_CMSSW_BASE}/lib/${SCRAM_ARCH}/.edmplugincache; do
+    if [ -e ${F} ] && [[ "${F}" != *"DataFormatsTestObjectsParent1"* ]]; then
+        ln -s ${F} ${CMSSW_BASE}/lib/${SCRAM_ARCH}
+    fi
+done
+
+# Symlink dictionary for edmtest::TransientIntParentT<1> from local/release
+# Also copy DataFormats/TestObjects code to be able to edit it to make ROOT header parsing to fail
+COPIED_FROM=
+for DIR in ${OLD_CMSSW_BASE} ${CMSSW_RELEASE_BASE} ${CMSSW_FULL_RELEASE_BASE} ; do
+    if [ -d ${DIR}/src/DataFormats/TestObjects ]; then
+        mkdir DataFormats
+        cp -r ${DIR}/src/DataFormats/TestObjects DataFormats/
+        cp ${DIR}/lib/${SCRAM_ARCH}/*DataFormatsTestObjectsParent1* ${CMSSW_BASE}/lib/${SCRAM_ARCH}
+        COPIED_FROM=${DIR}
+        break
+    fi
+done
+if [ ! -e ${CMSSW_BASE}/lib/${SCRAM_ARCH}/libDataFormatsTestObjectsParent1.so ]; then
+    echo "Failed to symlink libDataFormatsTestObjectsParent1.so from local or release area"
+    exit 1;
+fi
+if [ ! -e DataFormats/TestObjects ]; then
+    echo "Failed to copy DataFormats/TestObjects from local or release area"
+    exit 1;
+fi
+
+popd
+
+echo "Produce a file with TransientIntParentT<1> product"
+cmsRun ${LOCALTOP}/src/IOPool/Input/test/PoolNoParentDictionaryTestStep1_cfg.py || die 'Failed cmsRun PoolNoParentDictionaryTestStep1_cfg.py' $?
+
+
+# Then make attempt to load TransientIntParentT<1> to fail
+echo "PREVENT_HEADER_PARSING" >> ${CMSSW_BASE}/src/DataFormats/TestObjects/interface/ToyProducts.h
+rm ${CMSSW_BASE}/lib/${SCRAM_ARCH}/DataFormatsTestObjectsParent1_xr_rdict.pcm ${CMSSW_BASE}/lib/${SCRAM_ARCH}/libDataFormatsTestObjectsParent1.so
```

Comment:
@makortel , deleting these will not help as these files will be present in $CMSSW_RELEASE_BASE and root/cmsRun can find these from release area

Reply:
Right. But the trick on the following line (`sed` altering the shared object file in `DataFormatsTestObjectsParent1_xr.rootmap`) appears to make ROOT to not attempt to load the `DataFormatsTestObjectsParent1` stuff, and to attempt the header parsing. I just tested again, and apparently it is not even necessary to remove these files (i.e. the `sed` trick below alone is sufficient to trigger header parsing).